### PR TITLE
[TBDGen] Ensure symbols listed include maccatalyst when library is built zippered

### PR DESCRIPTION
**Pull request by cyndyishida**
This also introduces a dependency on new testing tool llvm-tapi-diff. This allows for a structural difference check on tbd files.

Resolves: rdar://107368040

### CODE REVIEW

**Review by xymus — APPROVED** [review]
This looks good!

File: test/TBD/zippered.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// REQUIRES: VENDOR=apple 
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+// RUN: %target-swift-frontend -target-variant %target-cpu-apple-ios13.1-macabi -typecheck -parse-as-library %t/test.swift -emit-tbd -emit-tbd-path %t/zippered.tbd
```
[the remark this answers is not in the document]
Comment (cyndyishida) [reply test/TBD/zippered.swift:5]: Oh, thanks for calling this out! I'd hope by this point the toolchains to build darwin would support macatalyst, but i'll add this to reduce potential fallout.

**Comment by cyndyishida** [comment]
@swift-ci please test

**Comment by cyndyishida** [comment]
@swift-ci please test

**Pull request merged**